pypesto/sampling/pymc3.py

```diff
@@ -112,11 +105,221 @@ def get_samples(self) -> McmcPtResult:
 
         if trace_x.shape[0] != trace_fval.shape[0] \
                 or trace_x.shape[1] != trace_fval.shape[1] \
-                or trace_x.shape[2] != len(self.problem.x_names):
+                or trace_x.shape[2] != len(self.problem.x_free_indices):
             raise ValueError("Trace dimensions are inconsistent")
 
         return McmcPtResult(
             trace_x=np.array(trace_x),
             trace_fval=np.array(trace_fval),
             betas=np.array([1.] * trace_x.shape[0]),
         )
+
+
+def create_pymc3_model(problem: Problem,
```

Comment:
where is this function used?

Reply:
if there is some magic that automatically converts this set of functions to a `PyMC3Model`, this should be documented somewhere and options testval, vectorize and lerp never seem to be used. I think we also would want some unittests for this function/class.

Reply:
I separated the pypesto interface from the creation of a pymc3 model, in order to make it easy to just use pymc3 directly for debugging/testing or if the user just wants to do so. In a previous commit I subclassed directly from `pymc3.Model`. It seemed to work, but after having a better look at it I noticed that such models seemed to be cached, so I could not create two of them with different `__init__` parameters. So I changed it to a function. I have not gone over the pypesto interface yet, so in the sampler class the old code was left over (sorry...)

Reply:
About doc/unit test will do.

Reply:
About unused parameters
* `testval` from the `pypesto` interface will be given as the initial sampling point if not restarting from a non-`None` trace (as mentioned in the issue head, I don't see how to easily restart sampling for `pymc3`)
* `vectorize` treats all parameters as a single vector. For normal `pymc3` models this is a huge speedup, when using `amici` not so much since I guess the ODE solver cost dominates everything else. I am not sure whether to expose this in the pypesto interface and what its default value should be, so input on this would be appreciated.
* `lerp` controls the formula for linear interpolation which is used in the `Interval` transform. I think here we should decide which one to use and not expose it to the pypesto user, as it is quite low level. The easy formula by convex combination is relatively safe, but is slightly less precise in many common-use cases and it is not monotone. The latter problem is probably not important in this case and possibly even the former slight precision loss can be accepted. There are ways to improve precision, but require taking a `max` or branching depending on the input value, so they are potentially slower (some quick-and-dirty testing tells me that they are slightly so, but better testing would be need). Again, if using `amici` any performance hit due to a different lerp formula would be drown out by the ODE solver, but not any `pypesto` user may be using `amici` in theory. I posted an issue on the `pymc3` tracker https://github.com/pymc-devs/pymc3/issues/3938 so if it is addressed we may then use the native solution.